### PR TITLE
feat: add robotframework-lsp support

### PR DESCRIPTION
**Pull request by egi**
Based on: https://github.com/neoclide/coc.nvim/wiki/Language-servers#robot-framework

### CODE REVIEW

File: lua/lspconfig/robotframework_ls.lua [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+local util = require 'lspconfig/util'
+
+local server_name = 'robotframework_ls'
+
+configs[server_name] = {
+  default_config = {
+    cmd = { 'robotframework_ls' },
+    filetypes = { 'robot' },
+    root_dir = function(fname)
+      return util.root_pattern '.git'(fname) or util.path.dirname(fname)
```
Comment (lithammer) [inline lua/lspconfig/robotframework_ls.lua:11]: We won't support `util.path.dirname(fname)` fallbacks until scratch server support has been implemented. However, I had a brief look at the robotframework-lsp and it seems like you'd want these root markers:

```python
    for directory in (common_base, *common_base.parents):
        if (directory / ".git").exists():
            return directory


        if (directory / "robotidy.toml").is_file():
            return directory


        if (directory / "pyproject.toml").is_file():
            return directory
```

https://github.com/robocorp/robotframework-lsp/blob/e2ea48e488840f592b20057f6049231d9cb05a01/robocorp-python-ls-core/src/robocorp_ls_core/libs/robotidy_lib/robotidy/files.py#L36-L44

Comment (lithammer) [reply lua/lspconfig/robotframework_ls.lua:11]:
```suggestion
      return util.root_pattern('pyproject.toml', 'robotidy.toml') or util.find_git_ancestor(fname)
```

**Comment by lithammer** [comment]
Squash the comments, then I'll merge.

File: lua/lspconfig/robotframework_ls.lua [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    end,
+  },
+  docs = {
+    description = [[
+https://github.com/robocorp/robotframework-lsp
+
+Language Server Protocol implementation for Robot Framework.
+]],
+    default_config = {
+      root_dir = "vim's starting directory",
```
Comment (mjlbach) [inline lua/lspconfig/robotframework_ls.lua:21]: Please update.

**Comment by egi** [comment]
thank you

**Pull request merged**